src/client/session/mod.rs

```diff
@@ -128,6 +360,12 @@ impl Drop for ClientSession {
             txn_number: self.server_session.txn_number,
         };
 
+        if self.transaction.state == TransactionState::InProgress {
+            RUNTIME.block_on(async {
```

Comment:
This was a bit tricky to implement, as the underlying functions called in `RUNTIME::execute` require the data within the `async move` block to have a `'static` lifetime. This isn't possible for `self` in this context because of the constraints of the signature of `drop`, so it seems like the only solution is to use `RUNTIME::block_on`. This doesn't seem great to me (since this blocks on the operation rather than executing it in the background) but I'm not sure there's a way to get around it -- do you have any better ideas?